src/applications/benefit-eligibility-questionnaire-all-benefits/containers/ConfirmationPage.jsx

```diff
@@ -0,0 +1,398 @@
+
+    const queryString = window.location.search;
+
+    this.state = {
+      hasResults: false,
+      resultsCount: 0,
+      benefitIds: [],
+      sortValue: 'alphabetical',
+      filterValue: 'All',
+      filterText: '',
+      benefits: [],
+      benefitsList: BENEFITS_LIST,
+      showMobileFilters: false,
+      currentQueryString: queryString,
+    };
+
+    this.applyInitialSort = this.applyInitialSort.bind(this);
+    this.sortBenefits = this.sortBenefits.bind(this);
+    this.filterBenefits = this.filterBenefits.bind(this);
+    this.createFilterText = this.createFilterText.bind(this);
+    this.toggleMobileFiltersClass = this.toggleMobileFiltersClass.bind(this);
+    this.filterAndSort = this.filterAndSort.bind(this);
+  }
+
+  componentDidMount() {
+    focusElement('h1');
+    scrollToTop('topScrollElement');
+  }
+
+  createFilterText() {
+    const resultsText = this.state.resultsCount === 1 ? 'result' : 'results';
+    return (
+      <>
+        Showing {this.state.resultsCount} {resultsText}, filtered to show{' '}
+        <b>{this.state.filterValue} results</b>, sorted{' '}
+        {this.state.sortValue === 'alphabetical'
+          ? 'alphabetically by benefit name'
+          : `alphabetically by benefit ${this.state.sortValue}`}
+      </>
+    );
+  }
+
+  handleSortSelect = e => {
+    const key = e.target.value;
+    const sortStrings = {
+      alphabetical: 'alphabetical',
+      category: 'type',
+    };
+    this.setState({ sortValue: sortStrings[key] });
+  };
+
+  sortBenefits = () => {
+    const key = this.state.sortValue;
+    const sortKey = key === 'alphabetical' ? 'name' : key;
+    this.setState(
+      prevState => {
+        if (!prevState.benefits || !Array.isArray(prevState.benefits)) {
+          return { benefits: [], benefitsList: [] };
+        }
+
+        const sortedBenefits = this.sortBenefitObj(prevState.benefits, sortKey);
+        const sortedBenefitsList = this.sortBenefitObj(
+          prevState.benefitsList,
+          sortKey,
+        );
+
+        return {
+          benefits: sortedBenefits,
+          benefitsList: sortedBenefitsList,
+        };
+      },
+      () => this.setState(() => ({ filterText: this.createFilterText() })),
+    );
+  };
+
+  handleFilterSelect = e => {
+    const key = e.target.value;
+
+    this.setState(() => ({ filterValue: key }));
+  };
+
+  filterBenefits = sortingCallback => {
+    const key = this.state.filterValue;
+    if (key === 'All') {
+      this.setState(
+        () => ({
+          benefitsList: BENEFITS_LIST,
+        }),
+        sortingCallback,
+      );
+      return;
+    }
+
+    this.setState(() => {
+      const filteredBenefitsList = BENEFITS_LIST.filter(benefit => {
+        return benefit.category.includes(key);
+      });
+      return {
+        benefitsList: filteredBenefitsList,
+      };
+    }, sortingCallback);
+  };
+
+  handleClick = e => {
+    e.preventDefault();
+
+    this.props.router.goBack();
```

Comment:
It looks like the new code at line 135, 139, and 140 introduces a slew of new linting warnings. Can we please clean those up before we merge this?